python/graphstorm/config/argument.py

```diff
@@ -1195,21 +1207,40 @@ def multilabel_weights(self):
 
            The weights should be in the following format 0.1,0.2,0.3,0.1,0.0
         """
-        if hasattr(self, "_multilabel_weights"):
-            assert self.multilabel is True, "Must be a multi-label classification task."
-            try:
-                weights = self._multilabel_weights.split(",")
-                weights = [float(w) for w in weights]
-            except Exception: # pylint: disable=broad-except
-                assert False, "The weights should in following format 0.1,0.2,0.3,0.1,0.0"
-            for w in weights:
-                assert w >= 0., "multilabel weights can not be negative values"
-            assert len(weights) == self.num_classes, \
-                "Each class must have an assigned weight"
-
-            return th.tensor(weights)
+        if hasattr(self, "_num_classes") and isinstance(self.num_classes, dict):
+            if hasattr(self, "_multilabel_weights"):
+                ntype_weights = {}
+                for ntype in self.num_classes:
+                    assert self.multilabel[ntype] is True, \
```

Comment:
should we save the `multilabel` result somewhere first? otherwise, it will trigger computations in multilabel every time.

Reply:
solved